### ISSUE TITLE
Incorrect horizontal alignment of tempo notation

### ISSUE THREAD

**Issue by Dima-S-Jr** [musescore/MuseScore#22579]
### Issue type

Engraving bug (incorrect score rendering)

### Bug description

Tempo notation is currently aligned to the left edge of the notehead. It should be borne in mind that alignment should be made along the left edge of not only the notehead, but also the musical element, if any (accidental, arpeggio, etc.)

### Steps to reproduce

1. Enter the notes
2. Add any tempo designation to the entered notes
3. Add accidentals to the entered notes (or any other elements placed before the note)

### Screenshots/Screen recordings

![MyCollages](https://github.com/musescore/MuseScore/assets/134041121/e9a84e15-1ceb-487b-b3a3-2f409c3a27c6)
![MyCollages (1)](https://github.com/musescore/MuseScore/assets/134041121/2dcba59b-5d9f-4f28-8d9d-978a9ad91229)
![MyCollages (2)](https://github.com/musescore/MuseScore/assets/134041121/7d761a5c-00e3-4b65-8096-74a4de4ffa27)

### MuseScore Version

MuseScore version (64-bit): 4.2.1-240230937, revision: github-musescore-musescore-d757433

### Regression

I don't know

### Operating system

Windows 11

### Additional context

This alignment was not adopted by chance. Agree that it is somehow inconvenient to see the musical elements first and then the tempo designation and the note itself. The same applies to the text of the staff/system/technical instructions/expression, etc. I proposed a similar alignment scheme in issue #21763, and I think it is quite reasonable.
When typing a score, I very often have to manually align a lot of text. And although there is a left alignment option in the properties, unfortunately, it is not taken into account that alignment should take into account the musical elements preceding the note. 
![21-11-33](https://github.com/musescore/MuseScore/assets/134041121/ff77f2b5-8cee-42d8-921d-a6090c2891ce)

**Comment by Dima-S-Jr** [comment]
> Wrong _in your opinion_. We're not doing this.

@oktophonie, I know you don't care how long it took me to write all this stuff, but let me clarify something. This is not my opinion, I am not the author of it. This is the opinion of the established practice of writing scores that has been developing for decades. Most music publishers who take music seriously and the rules of its writing adhere to exactly such a rule that tempo notation aligns to the left edge of the first musical element (mainly referring to the accidentals). This looks visually clearer and more elegant than when only the notehead and tempo symbols are aligned along the baseline, and the accidentals are placed outside the baseline, as if they are not included in the contents of the staffs and are only auxiliary elements (like staff brackets, for example).
![Снимок экрана (290)](https://github.com/musescore/MuseScore/assets/134041121/51df42e8-855d-41d8-8c13-f93804b6cb4d)
![Снимок экрана (289)](https://github.com/musescore/MuseScore/assets/134041121/aeaa7bb8-7c3a-47dc-81de-99ab61b4359a)
![Снимок экрана (288)](https://github.com/musescore/MuseScore/assets/134041121/ab43d6f8-443e-4e66-ae3c-a1c446648228)
![Снимок экрана (287)](https://github.com/musescore/MuseScore/assets/134041121/2314dff3-1629-4ca8-8568-708718c64c64)
I did not even give the relevant excerpts from the guids on musical notation. Alignment bypassing the accidentals looks like a hack. That's my opinion. 
However, if you are so unequivocal about this issue, then why didn't you add a "future request" label to this issue? Or are you against it at all, even if I create a separate issue as a "future request"?

**Comment by SnyBr** [comment]
(Github bug reports aren't the place for long discussions and debates, however, I feel a reply is warrented here. I'll try to be brief.)

In my opinion tempo markings should left-align with the noteheads. Here's why:

Noteheads' horizontal position marks the point in time at which they play. If something happens at that point in time, on the score it has to coincide with the given notehead. Following that logic, tempo text should coincide with the notehead. (Dynamics generally seem to follow a very similar logic, so why shouldn't tempo markings also do so?)

Doing it your way might even cause ambiguity in some cases. Your example illustrates this:

![Screenshot from 2024-04-26 22-41-05](https://github.com/musescore/MuseScore/assets/105515763/46d272e0-03a9-412f-920c-9a6f0df34d31)

Does the tempo mark apply to the 8th note? Or the arpeggiated halfnote chord? Or does the tempo change halfway through the 8th note? Does the 8th note suddenly slow down before its done playing?

As for publishers... afaik Henle is one of the most highly regarded publishers. They seem to favour aligning with the notehead. The following examples are from their editions of Chopin nocturnes and Liszt transcendental etudes.

![Screenshot from 2024-04-26 21-56-05](https://github.com/musescore/MuseScore/assets/105515763/483af8df-d1ee-4e6e-bb91-220dc3a146d7)
![tempo](https://github.com/musescore/MuseScore/assets/105515763/ac64c551-f201-408e-961d-2e983831d0e0)

From (seemingly good) scores made in Sibelius and Dorico I could quickly find online, it seems both these apps left-align to the notehead. I do think Dorico's opinion shouldn't be easily dismissed, since the team behind Dorico took it very seriously to achieve as good out-of-the-box engraving as they could.

So to conclude, expert opinion on this seems divided. At best its a house-style choice rather than a hard rule. And purely logically it makes more sense to me to align with the noteheads.

**Comment by Dima-S-Jr** [comment]
> (Github bug reports aren't the place for long discussions and debates, however, I feel a reply is warrented here. I'll try to be brief.)
> 
> In my opinion tempo markings should left-align with the noteheads. Here's why:
> 
> Noteheads' horizontal position marks the point in time at which they play. If something happens at that point in time, on the score it has to coincide with the given notehead. Following that logic, tempo text should coincide with the notehead. (Dynamics generally seem to follow a very similar logic, so why shouldn't tempo markings also do so?)
> 
> Doing it your way might even cause ambiguity in some cases. Your example illustrates this:
> 
> ![Screenshot from 2024-04-26 22-41-05](https://private-user-images.githubusercontent.com/105515763/326109008-46d272e0-03a9-412f-920c-9a6f0df34d31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQyMjA3NDgsIm5iZiI6MTcxNDIyMDQ0OCwicGF0aCI6Ii8xMDU1MTU3NjMvMzI2MTA5MDA4LTQ2ZDI3MmUwLTAzYTktNDEyZi05MjBjLTlhNmYwZGYzNGQzMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyN1QxMjIwNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNDYzZDkyM2Q1OTFkYWJlMjFlZjZkZjNkOWNlZjFkZTRjZjY0ZGMxZjkyOTQwMTUzZjYwYTIzMmQ2YmQyYjY3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NY-A90wF7qWfMkndk1ohitgvrvIalM41X5eA5KXzxYo)
> 
> Does the tempo mark apply to the 8th note? Or the arpeggiated halfnote chord? Or does the tempo change halfway through the 8th note? Does the 8th note suddenly slow down before its done playing?
> 
> As for publishers... afaik Henle is one of the most highly regarded publishers. They seem to favour aligning with the notehead. The following examples are from their editions of Chopin nocturnes and Liszt transcendental etudes.
> 
> ![Screenshot from 2024-04-26 21-56-05](https://private-user-images.githubusercontent.com/105515763/326111141-483af8df-d1ee-4e6e-bb91-220dc3a146d7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQyMjA3NDgsIm5iZiI6MTcxNDIyMDQ0OCwicGF0aCI6Ii8xMDU1MTU3NjMvMzI2MTExMTQxLTQ4M2FmOGRmLWQxZWUtNGU2ZS1iYjkxLTIyMGRjM2ExNDZkNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyN1QxMjIwNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YzNhYzc5NzgxMTk0ZTZkNzg4ZmEzM2FlMmE0N2ZhNDkxZjM0M2U4ZGY0YzYwOWU2NmJjODQ2NGI0ZTA0Zjk1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.JnxwioobK90yjIOu6DWHuEPpCBAnIcoaokCuzWI-SI8) ![tempo](https://private-user-images.githubusercontent.com/105515763/326113130-ac64c551-f201-408e-961d-2e983831d0e0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQyMjA3NDgsIm5iZiI6MTcxNDIyMDQ0OCwicGF0aCI6Ii8xMDU1MTU3NjMvMzI2MTEzMTMwLWFjNjRjNTUxLWYyMDEtNDA4ZS05NjFkLTJlOTgzODMxZDBlMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyN1QxMjIwNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02YTJhYTk1OTg4ZmNlNDFkMDgxOTZlNzlmMTk0YTNjMWUwNWYyNGI4OWI2Yjg0NDAwMjA0ZWFhOTU2ZWI2NWJjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.umj4zN1qVBGVEVMIbrZlJLhs7I9HTiXu_PPmbs2IayI)
> 
> From (seemingly good) scores made in Sibelius and Dorico I could quickly find online, it seems both these apps left-align to the notehead. I do think Dorico's opinion shouldn't be easily dismissed, since the team behind Dorico took it very seriously to achieve as good out-of-the-box engraving as they could.
> 
> So to conclude, expert opinion on this seems divided. At best its a house-style choice rather than a hard rule. And purely logically it makes more sense to me to align with the noteheads.

I agree that the example of the quaver is not entirely successful. However, different publishers have different opinions. Personally, my opinion is this: imagine that a pianist reads sheet music from a sheet. In the next beat, he first sees the accidentals, and, accordingly, he places his fingers accordingly. Then he sees the tempo designation (somewhere above), half the size of the previous one. This tempo designation is aligned to the left edge of the notehead of the quaver (that is, the performer must still look up before the note-value, and could have done this earlier if the tempo designation had been aligned to the left edge of the accidentals). This is unexpected for a pianist, since the tempo designation is more important, and I find it inconvenient to find out about it after placing my fingers (as if the one who wrote the sheet music meant: "Well, first of all, you will observe the accidentals, and then you will see the tempo"). But here it must be borne in mind that the tempo text is outside the staff, and not within its limits, and the performer's gaze should maneuver as follows: along the baseline - tempo / technical instructions / expression marks → alterations → noteheads.
As for programs such as Dorico, Sibelius and the music publisher you mentioned, I can say the following. Not all publishers adhere to meticulous alignments of each designation that go beyond the staff. I have come across such music publishers who, in one work, managed to align text designations in as many as three ways (by bar-line, by accidentals and by noteheads), which suggests that they did not adhere to any standard for text alignment. I can't speak for music editors like Sibelius and Dorico, but something tells me that in the settings of the musical notation style in these programs, text alignment settings for different types of musical elements should be enabled (especially in Sibelius, since there are a lot of settings there, and it's difficult for me to to sort out all the little things).
Summing up all of the above, I can say the following. Musescore should have a function for automatically aligning the text according to the musical elements that the user chooses. And yes, let those who hold the opinion that alignment should be done according to the noteheads, will have the opportunity to choose this method. And as Simon said that "_this is my opinion_" that in Musescore the alignment of the text by noteheads (and not by accidentals) is considered erroneous, I ask you to note that everyone who believes that this is not erroneous, then this is also "_your opinion_", since there is no such rule anywhere no, and if there is, then please provide proof. 
If this issue is not reopened, then I will have to create a similar one, just like a future request.

**Comment by oktophonie** [comment]
Please do, add it to the other 44 issues you've opened. We'll be sure to drop everything to get to it right away.

**Comment by SnyBr** [comment]
This will be my last reply on this thread.

1. Your thoughts on how a pianist's eyes glide along the score: No, your eyes instinctively jump to the noteheads. The noteheads are the instinctive "first stop" and centre of attention because a) visually they are the most attention-seeking (big boisterous black blobs) and b) they contain the most crucial information (which notes to play). And (without going into the nuances), the eyes+brain generally processes the chord+accidentals+stuff as a single unit in a single glimpse in the same way you read a word (or even a phrase) as a single unit and not its individual letters in sequence. There is no "first you read the accidentals, then this, then that". And I hope you're not saying that a pianist would know where to position his fingers before having seen the noteheads (assuming its the first time reading the score)?

2. Different publishers and notation apps doing different things: You note that not all publishers are meticulous with aligning stuff. This tells us not all publishers are equal in quality. But of those that are meticulous and deliberate in their alignment choices, we see they differ in their practices. House-style choice as I've said previously.

3. Giving users a choice between these two styles: I'm all for it, even though I disagree with your preference for reasons stated. But keep in mind, there are other priorities for what to do next in MuseScore. There is still plenty of engraving jank to fix, some of which require extensive overhauls under the hood. Getting all aspects of the current engraving up to a good standard is going to take priority above adding myriads of house-style options. So, if your feature-requests are going to be considered, it will likely not be soon. Which brings me to:

4. If you want your feature-requests to be taken more seriously:
a) Be mindful of your tone and attitude.
b) Make sure your arguments are well-informed and will withstand scrutiny.
c) Be open to correction.
d) Appeal-to-authority-style arguments ("this or that book/publisher/group says so") will rarely cut it. Instead of taking your chosen authorities' word for it, rather understand the "why's" and "why not's" and argue those.
e) Be cognizant of the wider development context (e.g. fixing jank before adding customization options, current engraving-overhauls in the pipeline, etc)
f) Be cognizant of how the team operates (keep noise on github to a minimum (I do apologise!), do long discussions on musescore.org (sorry again!), avoid duplicate issues/feature-requests, etc.)

This alignment-issue of yours as an example. You framed it as a right-vs-wrong thing when it actually is a house-style thing. This made you come across as both dogmatic and poorly informed.

You obviously have an eye for detail and plenty of useful observations to contribute. But you need to "integrate" (not precisely the word I'm looking for) better with the way things are done. That's why I'm taking the trouble of writing this wall of text.

I hope all of us can calm down a bit, get along and focus on the merits.

Okay, I'm done with my sermon and walls of text.
